syntax = "proto3";
package auth.v1;

service Authenticate {
  rpc Login (LoginReq) returns (LoginRes);
  rpc Refresh (RefreshReq) returns (AccessTokenRes);
}

service AccountService {
    rpc CreateAccount (AccountCreateReq) returns (AccountRes);
    rpc GetAccount (AccountGetReq) returns (AccountRes);
    rpc UpdateAccount (AccountUpdateReq) returns (AccountRes);
    rpc DeleteAccount (AccountDelReq) returns (AccountDelRes);
}

message LoginReq {
  string username = 1;
  string password = 2;
}

message LoginRes {
  string access = 1;
  string refresh = 2;
}

message RefreshReq {
    string refresh = 1;
}

message RefreshRes {
    string access = 1;
}


message AccountCreateReq {
    string login_id = 1;
    string password = 2;
}

message AccountGetReq {
    string account_id = 1;
}

message AccountUpdateReq {
    string account_id = 1;
    string login_id = 2;
    string password = 3;
}

message AccountDelReq {
    string account_id = 1;
}

message AccountRes {
    string account_id = 1;
    string login_id = 2;
    string password = 3;
}

message AccountDelRes {
    bool success = 1;
}
